numba/roc/hsaimpl.py

```diff
@@ -234,9 +235,20 @@ def hsail_atomic_add_tuple(context, builder, sig, args):
     return builder.atomic_rmw("add", ptr, val, ordering='monotonic')
 
 
-@lower('hsail.smem.alloc', types.UniTuple, types.Any)
-def hsail_smem_alloc_array(context, builder, sig, args):
-    shape, dtype = args
+@lower(roc.shared.array, types.IntegerLiteral, types.Any)
```

Comment:
This needs `from numba import roc` at the top.

Reply:
Added.